### PR TITLE
Add `ulid` support

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: src/Data/RdsData/Decode/Array.hs [hunk 1]
```diff
@@ -195,6 +198,13 @@ days = do
...
     Nothing -> DecodeArray \_ -> Left "Failed to decode Day"
 
+ulids :: DecodeArray [ULID]
+ulids = do
+  ts <- texts
+  case traverse CONV.textToUlid ts of
+    Right u -> pure u
+    Left msg -> DecodeArray \_ -> Left $ "Failed to decode UUID: " <> msg
+
 uuids :: DecodeArray [UUID]
```
Comment (Jimbo4350) [inline src/Data/RdsData/Decode/Array.hs:211]: Not necessary but I might put a comment by the encoder and decoder that steers consumers towards `ULID`s.

File: src/Data/RdsData/Internal/Convert.hs [hunk 2]
```diff
@@ -36,6 +42,16 @@ utcTimeToText :: UTCTime -> Text
...
   T.pack . formatTime defaultTimeLocale "%Y-%m-%d %H:%M:%S"
 
+ulidToText :: ULID -> Text
+ulidToText = 
+  ULID.encode 26 . ULID.ulidToInteger
+
+textToUlid :: Text -> Either Text ULID
+textToUlid t = 
+  case ULID.decode 26 t of
+    (n, ""):_ -> ULID.ulidFromInteger n
```
Comment (Jimbo4350) [inline src/Data/RdsData/Internal/Convert.hs]: Shouldn't we be pattern matching on the singleton list? Although it's only possible for this function to return `[]`or `[val]`.

**Review by Jimbo4350 — APPROVED** [review]

File: src/Data/RdsData/Internal/Convert.hs (hunk 2 above)
Comment (newhoggy) [reply src/Data/RdsData/Internal/Convert.hs]: I switched the code to use the `Read` instance with avoids having to deal with the list.

File: src/Data/RdsData/Decode/Array.hs (hunk 1 above)
Comment (newhoggy) [reply src/Data/RdsData/Decode/Array.hs:211]: Done

**Pull request merged**